backend/de.metas.material/dispo-commons/src/main/java/de/metas/material/dispo/commons/repository/CandidateRepositoryRetrieval.java

```diff
@@ -127,6 +130,8 @@
 	@VisibleForTesting
 	Optional<Candidate> fromCandidateRecord(final I_MD_Candidate candidateRecordOrNull)
 	{
+		final DimensionService dimensionService = SpringContextHolder.instance.getBean(DimensionService.class);
```

Comment:
since CandidateRepositoryRetrieval is also a spring service, i think you can inject DimensionService via constructor

Reply:
done